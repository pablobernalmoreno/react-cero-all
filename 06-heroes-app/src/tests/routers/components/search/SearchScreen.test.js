import { mount } from "enzyme";
import { MemoryRouter, Route } from "react-router-dom";
import { SearchScreen } from "../../../../components/search/SearchScreen";

describe("Pruebas en <SearchScreen/>", () => {
  test("debe de mostrarse correctamente con valores por defecto", () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={["/search"]}>
        <Route path="/search" component={SearchScreen} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".alert-info").text().trim()).toBe("Search a hero");
  });

  test("debe de mostrar a Batman y el 'input' con el valor del 'queryString'", () => {
    const hero = "Batman";
    const wrapper = mount(
      <MemoryRouter initialEntries={[`/search?q=${hero}`]}>
        <Route path="/search" component={SearchScreen} />
      </MemoryRouter>
    );

    expect(wrapper.find("input").prop("value")).toBe(hero);
  });

  test("debe de mostrar un error si no se encuentra el 'Hero'", () => {
    const hero = "Batman123123";
    const wrapper = mount(
      <MemoryRouter initialEntries={[`/search?q=${hero}`]}>
        <Route path="/search" component={SearchScreen} />
      </MemoryRouter>
    );

    expect(wrapper.find(".alert-danger").text().trim()).toBe(
      `There is no hero with name ${hero}`
    );
  });

  test("debe de llamar el 'push' del 'history'", () => {
    const history = {
      push: jest.fn(),
    };
    const hero = "Batman";
    const wrapper = mount(
      <MemoryRouter initialEntries={[`/search?q=${hero}`]}>
        <Route
          path="/search"
          component={() => <SearchScreen history={history} />}
        />
      </MemoryRouter>
    );

    wrapper.find("input").simulate("change", {
      target: {
        name: "searchText",
        value: hero,
      },
    });

    wrapper.find("form").prop("onSubmit")({
      preventDefault() {},
    });

    expect(history.push).toHaveBeenLastCalledWith(`?q=${hero}`);
  });
});
