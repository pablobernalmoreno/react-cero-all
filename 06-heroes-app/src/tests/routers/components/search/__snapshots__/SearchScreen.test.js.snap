// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchScreen/> debe de mostrarse correctamente con valores por defecto 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/search",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "utnng8",
            "pathname": "/search",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "utnng8",
          "pathname": "/search",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Route
      component={[Function]}
      path="/search"
    >
      <SearchScreen
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "utnng8",
                "pathname": "/search",
                "search": "",
                "state": undefined,
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "utnng8",
              "pathname": "/search",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "key": "utnng8",
            "pathname": "/search",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/search",
            "url": "/search",
          }
        }
      >
        <div>
          <h1>
            Search Screen
          </h1>
          <hr />
          <div
            className="row"
          >
            <div
              className="col-5"
            >
              <h4>
                Search Form
              </h4>
              <hr />
              <form
                onSubmit={[Function]}
              >
                <input
                  autoComplete="off"
                  className="form-control"
                  name="searchText"
                  onChange={[Function]}
                  placeholder="Find your hero"
                  type="text"
                  value=""
                />
                <button
                  className="btn m-1 btn-block btn-outline-primary"
                  type="submit"
                >
                  Search
                </button>
              </form>
            </div>
            <div
              className="col-7"
            >
              <h4>
                Results
              </h4>
              <hr />
              <div
                className="alert alert-info"
              >
                Search a hero
              </div>
            </div>
          </div>
        </div>
      </SearchScreen>
    </Route>
  </Router>
</MemoryRouter>
`;
